src/cli/configure/upgrade/patches/7.3.1/patchDockerfile.ts

```diff
@@ -8,8 +8,9 @@ import { createDestinationFileReader } from '../../../analysis/project';
 
 const DOCKERFILE_FILENAME = 'Dockerfile';
 
-const VERSION_REGEX = /gcr.io\/distroless\/nodejs:(16|18|20)/g;
-const VERSION_DEBIAN_REPLACE = 'gcr.io/distroless/nodejs$1-debian11';
+const NON_DEBIAN_REGEX = /gcr.io\/distroless\/nodejs:(16|18|20)/g;
+const DEBIAN_REGEX = /gcr.io\/distroless\/nodejs(16|18|20)-debian11/g;
```

Comment:
It looks like there is no `nodejs16-debian12` support: https://console.cloud.google.com/gcr/images/distroless

Reply:
Ooh that's complicated then. Maybe it should be two patches

Reply:
Ah, but skuba has engineConfig requiring 18 so this won't run on 16 anyway - maybe we just do the 18/20

Reply:
Oh nice one, sounds good 👍